src/agent/src/agent.cpp

```diff
@@ -10,11 +9,14 @@ Agent::Agent()
     m_taskManager.start(std::thread::hardware_concurrency());
     m_taskManager.enqueueTask(StatefulMessageProcessingTask(m_messageQueue));
     m_taskManager.enqueueTask(StatelessMessageProcessingTask(m_messageQueue));
+
+    int authRes = m_communicator.SendAuthenticationRequest();
```

Comment:
Don't do this here, this should be part of the CLI.

Reply:
Done